src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1055,30 +1070,57 @@ public override Task<object> ExecuteScalarAsync(CancellationToken cancellationTo
+            => (NpgsqlDataReader) base.ExecuteReader(behavior);
 
         /// <summary>
-        /// Executes the command text against the connection.
+        /// An asynchronous version of <see cref="ExecuteReader()"/>, which executes
+        /// the <see cref="CommandText"/> against the <see cref="Connection"/>
+        /// and returns a <see cref="NpgsqlDataReader"/>.
         /// </summary>
-        /// <param name="behavior">An instance of <see cref="CommandBehavior"/>.</param>
-        /// <param name="cancellationToken">A task representing the operation.</param>
-        /// <returns></returns>
-        protected override Task<DbDataReader> ExecuteDbDataReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public new Task<NpgsqlDataReader> ExecuteReaderAsync()
+            => ExecuteReaderAsync(CommandBehavior.Default, CancellationToken.None);
+
+        /// <summary>
+        /// An asynchronous version of <see cref="ExecuteReader()"/>, which executes
+        /// the <see cref="CommandText"/> against the <see cref="Connection"/>
+        /// and returns a <see cref="NpgsqlDataReader"/>.
+        /// </summary>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public new Task<NpgsqlDataReader> ExecuteReaderAsync(CancellationToken cancellationToken)
+            => ExecuteReaderAsync(CommandBehavior.Default, cancellationToken);
+
+        /// <summary>
+        /// An asynchronous version of <see cref="ExecuteReader(CommandBehavior)"/>,
+        /// which executes the <see cref="CommandText"/> against the <see cref="Connection"/>
+        /// and returns a <see cref="NpgsqlDataReader"/>.
+        /// </summary>
+        /// <param name="behavior">One of the enumeration values that specified the command behavior.</param>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public new Task<NpgsqlDataReader> ExecuteReaderAsync(CommandBehavior behavior)
+            => ExecuteReaderAsync(behavior, CancellationToken.None);
+
+        /// <summary>
+        /// An asynchronous version of <see cref="ExecuteReader(CommandBehavior)"/>,
+        /// which executes the <see cref="CommandText"/> against the <see cref="Connection"/>
+        /// and returns a <see cref="NpgsqlDataReader"/>.
+        /// </summary>
+        /// <param name="behavior">One of the enumeration values that specified the command behavior.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        public new Task<NpgsqlDataReader> ExecuteReaderAsync(CommandBehavior behavior, CancellationToken cancellationToken)
```

Comment:
Since this is a new API, I think we should have a default cancellation token here and merge with the overload above - as discussed in https://github.com/npgsql/npgsql/pull/2443/files#r317359536.